### PR TITLE
Meta-Parallelization, states and within states

### PR DESCRIPTION
**Pull request by fridiculous**
This PR tests meta-parallelization

### Please Check
- [x] Will the *current* schema in [api docs](https://github.com/covid-projections/covid-data-model/blob/master/api/README.md) be updated with this PR?
- [ ] Are tests passing?

### CODE REVIEW

File: pyseir/cli.py [hunk 1]
```diff
@@ -131,74 +163,189 @@ def _run_all(state=None, run_mode='default', generate_reports=True, output_inter
                 output_interval_days=output_interval_days,
                 skip_download=True,
                 states_only=states_only,
-                output_dir=output_dir
+                output_dir=output_dir,
             )
-            p = Pool()
-            p.map(f, [state_obj.name for state_obj in us.STATES])
-            p.close()
+            states_at_a_time = divide_up_pool()[1]
```
Comment (ghop02) [inline pyseir/cli.py:168]: _, states_at_a_time = divide_up_pool()

File: pyseir/__init__.py [hunk 2]
```diff
@@ -1,3 +1,40 @@
...
+    The latter integer is the number of states to run at the same time.
+
+    Returns:
+        (tuple[int, int]) -- a tuple of
+    """
+    total = os.cpu_count()
+    if total < 20:
+        return total, 1
+    # if total < 52:
+    return total, 10
```
Comment (ghop02) [inline pyseir/__init__.py:23]: Not sure I quite understand the split here.  Why does total not change?

File: pyseir/__init__.py [hunk 3]
```diff
@@ -1,3 +1,40 @@
...
+
+    def _set_daemon(self, value):
+        pass
+
+    daemon = property(_get_daemon, _set_daemon)
+
+
+# We sub-class multiprocessing.pool.Pool instead of multiprocessing.Pool
+# because the latter is only a wrapper function, not a proper class.
+class Pool(multiprocessing.pool.Pool):
```
Comment (ghop02) [inline pyseir/__init__.py:39]: Can you change the name of this to make it clear that it's not a normal pool?

Maybe `NoDaemonPool`

File: pyseir/__init__.py [hunk 4]
```diff
@@ -1,3 +1,40 @@
...
-OUTPUT_DIR = pathlib.Path(__file__).parent.parent / 'output'
+import os, math
+
+import multiprocessing.pool
+
+DATA_DIR = pathlib.Path(__file__).parent.parent / "data"
+OUTPUT_DIR = pathlib.Path(__file__).parent.parent / "output"
+
+
+def divide_up_pool():
```
Comment (ghop02) [inline pyseir/__init__.py:10]: Should this code live here? Seems like `libs/multiprocess_utils.py` or something might be more appropriate

File: pyseir/inference/model_fitter.py [hunk 5]
```diff
@@ -627,7 +629,9 @@ def run_state(state, states_only=False):
 
         df = load_data.load_county_metadata()
         all_fips = df[df['state'].str.lower() == state_obj.name.lower()].fips
-        p = Pool()
+
+        parallelization_within_states = divide_up_pool()[0]
```
Comment (ghop02) [inline pyseir/inference/model_fitter.py:633]: `state_parallelization, _ = divide_up_pool()`

File: pyseir/__init__.py (hunk 4 above)
Comment (ghop02) [inline pyseir/__init__.py:10]: could also split these into two functions.  `divide_up_pool` doesn't make it clear that it's picking the pool size of county and state level runs. 
```
def get_state_pool_size():

def get_county_pool_size():
```

**Comment by fridiculous** [comment]
Closing as looking at a new strategy to break up the code

**Pull request closed without merging**